### ISSUE TITLE
Feature request: support `doctrine/cache:^2.0`

### ISSUE THREAD

**Issue by ruscon** [Dropelikeit/laravel-jms-serializer#19]
### Description of the problem:

```bash
✗ sail composer require dropelikeit/laravel-jms-serializer                                     
Info from https://repo.packagist.org:
Using version ^3.0 for dropelikeit/laravel-jms-serializer
./composer.json has been updated
Running composer update dropelikeit/laravel-jms-serializer
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires dropelikeit/laravel-jms-serializer ^3.0 -> satisfiable by dropelikeit/laravel-jms-serializer[v3.0.0].
    - dropelikeit/laravel-jms-serializer v3.0.0 requires doctrine/cache ^1.10 -> found doctrine/cache[1.10.0, ..., 1.13.x-dev] but the package is fixed to 2.2.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
You can also try re-running composer require with an explicit version constraint, e.g. "composer require dropelikeit/laravel-jms-serializer:*" to figure out if any version is installable, or "composer require dropelikeit/laravel-jms-serializer:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

```

**Comment by Dropelikeit** [comment]
Hello @ruscon,

thank you for reporting this error. I have looked into the reported issue and was able to fix it.

I have removed the complete dependency to `doctrine/cache` in the pr, as the `doctrine/cache` library is deprecated and should not be used anymore. This library needed the `doctrine/cache` library to make the tests work, so I added `symfony/cache` as a dev dependency to ensure that the functionality of the tests is preserved.
In my opinion, the library should not bring a cache library itself, but use the psr-6 / psr-16 compatible cache library from the project.
